lib/build/geo_feature_builder.dart

```diff
@@ -0,0 +1,155 @@
+      } 
+      finalGeoFeatureList.add(extractedProperties);
+    }
+  }
+
+  String convertPolygonData(GeoJsonPolygon geoJsonPolygon){
+    late List<String> outer = [];
+      final List<String> inner = [];
+      for (int index = 0; index < geoJsonPolygon.geoSeries.length; index++) {
+        final List<String> points = [];
+        for (final geopoint in geoJsonPolygon.geoSeries[index].geoPoints){
+          points.add('LatLng(${geopoint.latitude},${geopoint.longitude})');
+        }
+        index == 0 ? outer = points : inner.add('Ring($points)');
+      }
+    return 'Polygon(Ring($outer),$inner)';
+  }
+
+  String createClass(){
+
+    final List<Field> classFields = [];
+    final List<Parameter> constructorParameters = [];
+
+    propertiesToExtract.forEach((key, value) { 
+      classFields.add(Field((b) => b
+        ..modifier = FieldModifier.final$
+        ..name = ReCase(key).camelCase 
+        ..type = Reference(value,'dart:core').type));
+      constructorParameters.add(Parameter((b) => b
+          ..name = ReCase(key).camelCase
+          ..named = true
+          ..toThis = true));
+    });
+    constructorParameters.add(Parameter((b) => b
+            ..name = 'area'
+            ..named = true
+            ..toSuper = true
+    ));
+
+    final geoFeatureClass = ClassBuilder()
+      ..name = 'GeoFeature'
+      ..extend = refer('GeoFeatureBase')
+      ..fields.addAll(classFields)
+      ..constructors.add(Constructor((b) => b
+        ..constant = true
+        ..requiredParameters.addAll(constructorParameters)
+      ));
+
+    final geoFeatureCollection = finalGeoFeatureList.map((e) {
+      var constructor = '';
+      for (final key in propertiesToExtract.keys) {
+          final propertyValue = propertiesToExtract[key] == 'String' 
+          ? '\'${e[ReCase(key).camelCase].toString()}\',' 
+          : '${e[ReCase(key).camelCase].toString()},';
+           constructor = '$constructor $propertyValue';
+      }
+      constructor = '$constructor ${e['area']}';
+      return 'GeoFeature($constructor)';
+    }).toList();
+
+    final geoFeatureCollectionClass = ClassBuilder()
+      ..name = 'GeoFeatureCollection'
+      ..extend = refer('GeoFeatureCollectionBase')
+      ..constructors.add(Constructor((b) => b))
+      ..fields.add(Field((b) => b
+        ..name = 'geoFeatureCollection'
+        ..modifier = FieldModifier.final$
+        ..assignment = Code(geoFeatureCollection.toString())
+        ..type = refer('List<GeoFeature>')
+      ));
+
+    final library = Library((b) => b
+      ..directives.add(Directive.import('package:offline_geocoder/offline_geocoder.dart'))
+      ..directives.add(Directive.import('package:latlong2/latlong.dart'))
+      ..body.addAll([geoFeatureClass.build(), geoFeatureCollectionClass.build()]));
+
+    final dartfmt = DartFormatter();
+    return dartfmt.format('${library.accept(DartEmitter.scoped())}').toString();
+  } 
+}
```

Comment:
Requires some re-factoring.

Definitely try to make this code more readable. Split it into meaningfull functions. The Class Builders for `GeoFeature` and `GeoFeatureCollection` (new name `GeoCoder`) for example should be separated into two functions instead of thrown into one big block of code.

```dart
Class buildGeoFeature(Map<String, String> fieldDefinitions) {
  final classFields = fieldDefinitions.entries.map((entry) {
    classFields.add(Field((b) => b
      ..modifier = FieldModifier.final$
      ..name = ReCase(entry.key).camelCase
      ..type = Reference(entry.value, 'dart:core').type,
    ));
  });
  
  final constructorParameters = fieldDefinitions.entries.map((entry) {
    constructorParameters.add(Parameter((b) => b
      ..name = ReCase(entry.key).camelCase
      ..named = true
      ..toThis = true,
    ));
  }).followedBy([
    Parameter((b) => b
      ..name = 'area'
      ..named = true
      ..toSuper = true,
    ),
  ]);

  return ClassBuilder()
    ..name = 'GeoFeature'
    ..extend = refer('GeoFeatureBase')
    ..fields.addAll(classFields)
    ..constructors.add(Constructor((b) => b
      ..constant = true
      ..requiredParameters.addAll(constructorParameters)
    ));
}


Class buildGeoCoder(anyDataYouNeed) {
   ...
}
```